backend/app/routes/web/research_product.py

```diff
@@ -0,0 +1,57 @@
+"""Endpoint for adding a research product to favourites"""
+import asyncio
+import logging
+from contextlib import suppress
+from typing import List, Optional
+
+from fastapi import APIRouter, Depends
+from httpx import AsyncClient, ConnectError
+from pydantic import AnyUrl
+
+from app.schemas.research_product_response import Collection, ResearchProductResponse
+from app.solr.operations import get_dep
+
+router = APIRouter()
+
+logger = logging.getLogger(__name__)
+
+
+@router.get("/research-product/{resource_type}/{rp_id}")
+async def get_rp_by_id(
+    resource_type: Collection, rp_id: str, solr_get=Depends(get_dep)
+) -> ResearchProductResponse:
+    """
+    Main function responsible for getting details for a given Solr document.
+    args:
+        resource_type: string literal - one of permitted collections
+        rp_id: ID of a research product to be found
+    """
```

Comment:
Docstring here will be visible in swagger - so I'd advice top phrase it more like it would be in API docs. Especially args. If we have literals specyfied here then swagger will correctly show which values are acceptable.